src/test/java/io/zold/api/NetworkTest.java

```diff
@@ -0,0 +1,119 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.zold.api;
+
+import java.util.ArrayList;
+import org.cactoos.iterable.IterableOf;
+import org.cactoos.iterable.Repeated;
+import org.cactoos.text.RandomText;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Test case for {@link Network}.
+ *
+ * @author Paulo Lobo (pauloeduardolobo@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @todo #5:30min Implement Remote interface. Remote Interface must be
+ *  implemented because Network depends on Remote behavior. Network.pull
+ *  needs to search all remotes for some wallet id and merge all found
+ *  wallets; Network.push must add a wallet to a remote based in remote.
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+public final class NetworkTest {
+
+    /**
+     * Network can push wallet to right remote.
+     * Tests if the {@link Network} pulls the {@link Wallet} to the remote
+     * selected according rules defined in {@link Network}: {@link Remote}
+     * with higher score and ({@link Remote#score()} > 16. Throws a
+     * {@link RuntimeException} if {@link Network} tries to push the wallet
+     * to the wrong node.
+     */
+    @Test
+    public void pullWalletToRightRemote()  {
+        final Remote highremote = Mockito.mock(Remote.class);
+        final Score highscore = Mockito.mock(Score.class);
+        Mockito.when(highscore.suffixes()).thenReturn(
+            new Repeated<>(20, new RandomText())
+        );
+        Mockito.when(highremote.score()).thenReturn(highscore);
+        final Remote mediumremote = Mockito.mock(Remote.class);
+        final Score mediumscore = Mockito.mock(Score.class);
+        Mockito.when(mediumscore.suffixes()).thenReturn(
+            new Repeated<>(10, new RandomText())
+        );
+        Mockito.when(mediumremote.score()).thenReturn(mediumscore);
+        Mockito.doThrow(new RuntimeException()).when(mediumremote).add(
+            Mockito.any(Wallet.class)
+        );
+        final Remote lowremote = Mockito.mock(Remote.class);
+        final Score lowscore = Mockito.mock(Score.class);
+        Mockito.when(lowscore.suffixes()).thenReturn(
+            new Repeated<>(5, new RandomText())
+        );
+        Mockito.when(lowremote.score()).thenReturn(lowscore);
+        Mockito.doThrow(new RuntimeException()).when(lowremote).add(
+            Mockito.any(Wallet.class)
+        );
+        final Wallet wallet = Mockito.mock(Wallet.class);
+        new Network.Simple(
+            new IterableOf<Remote>(
+                highremote, mediumremote, lowremote
```

Comment:
@paulodamaso two remotes should be enough here. Also, let's make sure both remotes have a score above 16. The way your test is set up now, two of the remotes are filtered because of their score, even before they are ranked.

Reply:
@carlosmiranda You are right, let's use two remotes, I'll fix it thanks.